common/common.go

```diff
@@ -610,22 +610,40 @@ func GetDefaultDataDir(env string) string {
 	return path.Join(os.ExpandEnv("$HOME"), ".gocryptotrader")
 }
 
-// CheckDir checks to see if a particular directory exists
-// and attempts to create it if desired, if it doesn't exist
-func CheckDir(dir string, create bool) error {
+// CreateDir creates a directory based on the supplied parameter
+func CreateDir(dir string) error {
 	_, err := os.Stat(dir)
 	if !os.IsNotExist(err) {
 		return nil
 	}
 
-	if !create {
-		return fmt.Errorf("directory %s does not exist. Err: %s", dir, err)
-	}
-
 	log.Warnf("Directory %s does not exist.. creating.", dir)
-	err = os.MkdirAll(dir, 0777)
+	return os.MkdirAll(dir, 0770)
+}
+
+// ListAllDir lists all the directories and files in an array
+func ListAllDir(directory string) ([]string, error) {
+	var list []string
+	err := filepath.Walk(directory, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			list = append(list, path)
+		}
+		return nil
+	})
+	return list, err
+}
+
+// ChangePerm changes file permissions in a given directory to 0770
+func ChangePerm(env string) error {
+	a, err := ListAllDir(GetDefaultDataDir(env))
 	if err != nil {
-		return fmt.Errorf("failed to create dir. Err: %s", err)
+		return errors.New("Directory provided is invalid")
+	}
+	for i := range a {
+		b, _ := os.Stat(a[i])
+		if b.Mode().Perm() != 0770 {
+			os.Chmod(a[i], 0770)
```

Comment:
chmod does return an error that is worth checking to confirm there was no issues setting the permissions requested